react/lib/util/cashtab.ts

```diff
@@ -0,0 +1,100 @@
+import { 
+  CashtabConnect, 
+  CashtabExtensionUnavailableError,
+  CashtabAddressDeniedError,
+  CashtabTimeoutError 
+} from 'cashtab-connect';
+
+// Create a single instance to be reused throughout the app
+const cashtab = new CashtabConnect();
+
+/**
+ * Check if the Cashtab extension is available
+ * @returns Promise<boolean> - true if extension is available, false otherwise
+ */
+export const getCashtabProviderStatus = async (): Promise<boolean> => {
+  try {
+    return await cashtab.isExtensionAvailable();
+  } catch (error) {
+    return false;
+  }
+};
+
+/**
+ * Wait for the Cashtab extension to become available
+ * @param timeout - Maximum time to wait in milliseconds (default: 3000)
+ * @returns Promise that resolves when extension is available or rejects on timeout
+ */
+export const waitForCashtabExtension = async (timeout?: number): Promise<void> => {
+  return cashtab.waitForExtension(timeout);
+};
+
+/**
+ * Request the user's eCash address from their Cashtab wallet
+ * @returns Promise<string> - The user's address
+ * @throws {CashtabExtensionUnavailableError} When the Cashtab extension is not available
+ * @throws {CashtabAddressDeniedError} When the user denies the address request
+ * @throws {CashtabTimeoutError} When the request times out
+ */
+export const requestCashtabAddress = async (): Promise<string> => {
+  return cashtab.requestAddress();
+};
+
+/**
+ * Send XEC to an address using Cashtab
+ * @param address - Recipient's eCash address
+ * @param amount - Amount to send in XEC
+ * @throws {CashtabExtensionUnavailableError} When the Cashtab extension is not available
+ */
+export const sendXecWithCashtab = async (address: string, amount: string | number): Promise<any> => {
+  return cashtab.sendXec(address, amount);
+};
+
+/**
+ * Open Cashtab with a BIP21 payment URL
+ * @param bip21Url - The BIP21 formatted payment URL
+ * @param fallbackUrl - Optional fallback URL if extension is not available
+ */
+export const openCashtabPayment = async (bip21Url: string, fallbackUrl?: string): Promise<void> => {
+  try {
+    const isAvailable = await getCashtabProviderStatus();
```

Comment:
this approach -- checking status on button click -- is probably effective in most current contexts. However it would be wasteful for any application where multiple clicks are expected.

The cashtab provider status is not expected to change during a user's visit (and, even if it did --- e.g. if the user installs the cashtab extension --- a refresh is required for this to be detected).

Would be more robust and payments would be faster if this check were performed on page load. This would also allow you to conditionally render the button, i.e. show the user that it will open Cashtab or the mobile app or Electrum before a click.

Reply:
Good idea - now we can make this nice little text change on widget load:

<img width="753" height="480" alt="image" src="https://github.com/user-attachments/assets/50331fc3-1734-45f9-9884-45db80011cff" />